tools/pylib/boutdata/squashoutput.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+
+"""
+Collect all data from BOUT.dmp.* files and create a single output file.
+
+Output file named BOUT.dmp.nc by default
+
+Useful because this discards ghost cell data (that is only useful for debugging)
+and because single files are quicker to download.
+
+When run as script:
+    - first command line argument specifies data directory (default is the
+      current directory where the script is run)
+    - optional argument "--outputname <name>" can be given to change the name
+      of the output file
+"""
+
+from boutdata.data import BoutOutputs
+from boututils.datafile import DataFile
+from boututils.boutarray import BoutArray
+import numpy
+import os
+
+def squashoutput(datadir=".", outputname="BOUT.dmp.nc", format="NETCDF4", tslice=[None,None,None], xslice=[None,None,None], yslice=[None,None,None], zslice=[None,None,None], singleprecision=False):
```

Comment:
The default argument for `tslice` etc. should just be `None` rather than a list of `None`s. See [here][1] for why. Note that you'll need to modify your checks later.


[1]: http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments